x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.ts

```diff
@@ -56,7 +62,7 @@ export class EnterpriseSearchRequestHandler {
     this.enterpriseSearchUrl = config.host as string;
   }
 
-  createRequest({ path, params = {}, hasValidData = () => true }: RequestParams) {
+  createRequest({ path, params = {}, hasValidData = () => true, computeExtraParams = () => {} }: RequestParams) {
```

Comment:
It feels a little weird to me to have two different methods for adding extra params. I'm tempted to survey where the `params` option is used and consider combining the two.

Reply:
@constancecchen set me straight on how to make use of the existing params method, so I've removed this additional one.